x-pack/plugins/endpoint/server/routes/whitelist.ts

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter, RequestHandlerContext, APICaller } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { SearchResponse } from 'elasticsearch';
+import { createHash } from 'crypto';
+
+import lzma from 'lzma-native';
+import { WhitelistRule, WhitelistSet } from '../../common/types';
+import { EndpointAppContext } from '../types';
+
+const whitelistIdx: string = 'whitelist'; // TODO: change this
+const allowlistBaseRoute: string = '/api/endpoint/allowlist';
+const whitelistLimit: number = 1000;
+let whitelistArtifactCache: Buffer = Buffer.from([]);
```

Comment:
Few design questions:
* Why is this cache needed?
* Will this work as expected if the customer has several Kibana instances running horizontally against the same ES cluster?

Reply:
We need to cache the whitelist so when the n sensors call in to download it, we don't have to query elasticsearch and comrpess n times to get the same data.

I need to test it on a multi-kibana set up (I haven't been successful in setting one up on my own). Have you been able to stand one up?